### PR TITLE
Fix CircleCI

### PR DESCRIPTION
**Pull request by helicopter88**
CircleCI does now run, but the tests that are failing locally also fail on circle

### CODE REVIEW

File: src/klee_web/frontend/views.py [hunk 1]
```diff
@@ -37,6 +36,7 @@ def jobs_notify(request):
     return HttpResponse('Ok!')
 
 
+
```
Comment (andronat) [inline src/klee_web/frontend/views.py]: Remove this extra line. methods are separated with only 2 empty lines not 3.

**Comment by helicopter88** [comment]
If you check the build time, for some reasons, circle gets stuck for a lot of time at installing the tests dependencies, I just checked the provisioning file and there is nothing strange there

**Comment by andronat** [comment]
What about?

```
flake8 --max-complexity 12 --exclude=migrations /home/ubuntu/klee-web
/home/ubuntu/klee-web/src/klee_web/frontend/views.py:23:2: F821 undefined name 'require_POST'
```

For `TASK [webdriver : Install E2E test dependencies]` is it working locally to you?

**Comment by helicopter88** [comment]
I just fixed that, basically it got removed in the other PR and forgot to bring it back

The installation E2E dependencies don't take much time on my laptop

**Comment by andronat** [comment]
If it fails again give a verbose flag to ansible to print everything so we can see what's going on in circleci.

**Comment by andronat** [comment]
Try also with `-vvv` and `-vvvv`. If we still don't see anything useful try and remove version dependences in the [package file](https://github.com/klee/klee-web/blob/master/src/klee_web/tests/package.json). Maybe something is stuck there. As a last recommendation maybe you can run the playbook locally in vagrant with the `ci` variable enabled to replicate exactly what circleci runs locally. If everything works at your side we might have to think that the problem is with circleci and not with us (which most probably is not the case).

**Comment by andronat** [comment]
So it seems like the problem is with the versions. It would be great if you find which one has the problem and also see for each of the dependences we would update in general. It is not always good to have unrestricted versioning to dependences as a new version come out which breaks some functionality. We should do a general maintenance in these packages.

**Comment by andronat** [comment]
BTW, I also see that `run_tests.sh` reports a 500 error. Do you know why?

**Comment by helicopter88** [comment]
The error is there because python still tries to use Pusher in this PR, but fails because of no key. JS needs the pusher key too and that may cause failures too

**Comment by andronat** [comment]
Ok i think this is good. So let's fix the dependences and we can merge this one.

**Comment by andronat** [comment]
Great! Just one last issue. We have the same dependences installed in two different places: 
1. mocha [1](https://github.com/klee/klee-web/blob/ece9745b1556a7c468688325a7ee1ccf9dac12cf/provisioning/roles/webdriver/tasks/main.yml#L30) & [2](https://github.com/klee/klee-web/pull/75/files#diff-72521b1585504d9f1a92f03c211c5518R7)
2. phantomjs [1](https://github.com/klee/klee-web/blob/ece9745b1556a7c468688325a7ee1ccf9dac12cf/provisioning/roles/webdriver/tasks/main.yml#L26) & [2](https://github.com/klee/klee-web/pull/75/files#diff-72521b1585504d9f1a92f03c211c5518R9)

Is this is required? For sure it is not safe as imagine the playbook installing a different version than `packages.json` what happens then? Can we keep only 1 of them?

Also, inside the same playbook we have [1](https://github.com/klee/klee-web/blob/ece9745b1556a7c468688325a7ee1ccf9dac12cf/provisioning/roles/webdriver/tasks/main.yml#L21) & [2](https://github.com/klee/klee-web/blob/ece9745b1556a7c468688325a7ee1ccf9dac12cf/provisioning/roles/webdriver/tasks/main.yml#L30). Is this necessary?

**Comment by helicopter88** [comment]
We need them in the yml because they're global, and run_tests depends on that, now I am not sure if we need them in package.json
Not sure why they're there twice, will clean it up

**Comment by helicopter88** [comment]
No, they're not needed in package.json, but they're needed twice in the yml because of the become: no, as it does not work with vagrant, but it does work on circle(and vice versa, become: yes does not work on circle)

**Comment by andronat** [comment]
From the official manual of Mocha I found [this](http://www.scotchmedia.com/tutorials/express/authentication/1/03) and from googling [this](http://stackoverflow.com/questions/20013989/configure-node-npm-package-json-so-that-npm-test-works-on-both-unix-and-window). They explain how to use `npm` and overcome the need to install Mocha globally. Can we make the proper alternations to get rid of global installations? Also can you check if we can do the same for `phantomjs`?

In general the logic is that everything that has to do with javascript and `node.js` should be managed per-project (inside a package.json) and we should not have global dependences.

Lastly, did you find why there is this difference of behaviour between Vagrant and CircleCI? It doesn't make sense to me that Vagrant doesn't allow you to work as a local user.

**Comment by helicopter88** [comment]
So I have removed both phantomjs and mocha from being globally installed and the tests seem to fail in the exact same way.
I haven't looked at why vagrant and circle do not behave in the same way however, but with the latest commit there is no need to worry about that

File: src/klee_web/tests/package.json [hunk 2]
```diff
@@ -2,11 +2,16 @@
   "name": "E2E-Tests",
   "description": "End-To-End tests for KLEE Web",
   "version": "0.0.1",
+  "scripts": {
+    "test": "node_modules/.bin/mocha /titb/src/klee_web/tests/e2e_tests.js"
```
Comment (andronat) [inline src/klee_web/tests/package.json]: Is this working? I don't think this is a valid syntax. Did you try to run the tests locally first?

Comment (andronat) [inline src/klee_web/tests/package.json]: Ignore my comment. I read it wrongly.

**Comment by andronat** [comment]
Ok last think. I see that actually we don't use `phantomjs` anywhere... Can you verify that too? I see that another library is used instead `nightmare`. If we don't have any use for `phantomjs` remove it.

**Comment by andronat** [comment]
In the end also increase the version in `package.json`

File: src/klee_web/tests/package.json (hunk 2 above)
Comment (andronat) [inline src/klee_web/tests/package.json]: Can you try to make the absolute path `/titb/src/klee_web/tests/e2e_tests.js`, relative? `./e2e_tests.js`. I would be nice not to have path dependences.

**Comment by helicopter88** [comment]
Tests work as usual locally

**Comment by helicopter88** [comment]
Turns out the test suite does not work with newer nightmare, so for now, using the old one + phantomjs solves the issue.
On a side node, this only works if we disable websecurity for nightmare

**Comment by ccadar** [comment]
Hi, what's the status of CircleCI?  This PR suggests it has been fixed, but I get an error when I access https://circleci.com/gh/klee-web/klee-web

**Comment by andronat** [comment]
@ccadar for some reason the current repository is synced with @helicopter88 CircleCI [here](https://circleci.com/gh/helicopter88/klee-web).

**Pull request merged**